tests/setup_teardown.rs

```diff
@@ -26,130 +46,334 @@ pub async fn get_index(user: &GooseUser) -> GooseTaskResult {
+    endpoints.push(
+        Mock::new()
+            .expect_method(GET)
+            .expect_path(INDEX_PATH)
+            .return_status(201)
+            .create_on(&server),
+    );
+    // Next set up SETUP_PATH, store in vector at SETUP_KEY.
+    endpoints.push(
+        Mock::new()
+            .expect_method(POST)
+            .expect_path(SETUP_PATH)
+            .return_status(205)
+            .create_on(&server),
+    );
+    // Next set up TEARDOWN_PATH, store in vector at TEARDOWN_KEY.
+    endpoints.push(
+        Mock::new()
+            .expect_method(POST)
+            .expect_path(TEARDOWN_PATH)
+            .return_status(200)
+            .create_on(&server),
+    );
+
+    endpoints
+}
+
+// Create a custom configuration for this test.
+fn common_build_configuration(
+    server: &MockServer,
+    worker: Option<bool>,
+    manager: Option<usize>,
+) -> GooseConfiguration {
+    if let Some(expect_workers) = manager {
+        common::build_configuration(
+            &server,
+            vec![
+                "--manager",
+                "--expect-workers",
+                &expect_workers.to_string(),
+                "--users",
+                USERS,
+                "--hatch-rate",
+                USERS,
+            ],
+        )
+    } else if worker.is_some() {
+        common::build_configuration(&server, vec!["--worker"])
+    } else {
+        common::build_configuration(&server, vec!["--users", USERS, "--hatch-rate", USERS])
+    }
+}
+
+// Common validation for the load tests in this file.
+// @FIXME: remove the `is_gaggle` flag once issue #182 lands.
+fn validate_test(test_type: &TestType, mock_endpoints: &Vec<MockRef>, is_gaggle: bool) {
+    // Confirm the load test ran.
+    assert!(mock_endpoints[INDEX_KEY].times_called() > 0);
+
+    // Now confirm TestType-specific counters.
+    if is_gaggle {
+        // @FIXME: when https://github.com/tag1consulting/goose/issues/182 lands
+        // remove the `is_gaggle` flag and everything in it: the gaggle and standalone
+        // tests should have the same validation.
+        match test_type {
+            TestType::Start => {
+                // @FIXME: setup should have run.
+                assert!(mock_endpoints[SETUP_KEY].times_called() == 0);
+                assert!(mock_endpoints[TEARDOWN_KEY].times_called() == 0);
+            }
+            TestType::Stop => {
+                // @FIXME: teardown should have run.
+                assert!(mock_endpoints[SETUP_KEY].times_called() == 0);
+                assert!(mock_endpoints[TEARDOWN_KEY].times_called() == 0);
+            }
+            TestType::StartAndStop => {
+                // @FIXME: setup and teardown should have run.
+                assert!(mock_endpoints[SETUP_KEY].times_called() == 0);
+                assert!(mock_endpoints[TEARDOWN_KEY].times_called() == 0);
+            }
+        }
+    } else {
+        match test_type {
```

Comment:
Only the `else` block will remain when we can remove `is_gaggle`, as the test should have identical results whether running in standalone or gaggle mode.